gems/sorbet-runtime/lib/types/props/custom_type.rb

```diff
@@ -62,7 +62,8 @@ def self.scalar_type?(val)
       # We don't need to check for val's included modules in
       # T::Configuration.scalar_types, because T::Configuration.scalar_types
       # are all classes.
-      klass = T.let(val.class, T.nilable(Class))
+      # We elide `klass : T.nilable(Class)` to avoid allocations.
+      klass = T.unsafe(val.class)
```

Comment:
Another option is to mark `val` as having type `T.untyped` if you want to save the cost of the call to `T.unsafe` too.

Note that external callers of `scalar_type?` (anyone calling this outside of the sorbet-runtime gem) already see `val` as `T.untyped` because the sig on this method was not mirrored into `rbi/sorbet/tprops.rbi`.

Reply:
Thanks, that sounds even better!